eval_protocol/pytest/default_single_turn_rollout_process.py

```
@@ -112,6 +112,8 @@ async def _sem_wrapper(r: EvaluationRow) -> EvaluationRow:
             try:
                 return await process_row(r)
             except Exception as e:
+                r.rollout_status.status = "error"
+                r.rollout_status.termination_reason = str(e)
```

I actually think this try/except logic shouldn't live inside the rollouts. The retry logic and handling should happen all outside of the rollout to separate concerns

---

we want to eventually implement this ticket right: https://linear.app/fireworks/issue/FIR-4431/implement-exception-handling-by-using-python-backoff-library

That means the native Python exception handling should happen outside of the rollout processor. Right now we don't have that information if you just catch all exceptions and use `rollout_status` to communicate the error happened

---

we should ideally remove the `try/except` altogether in the rollouts and have the outer `evaluation_test` handle everything

---

gotcha, the intention here is because from yinghan's changes, the `default_mcp_gym_rollout_processor.py` is what handles the `rollout_status`, so i thought same should apply to single turn and agent. given that, should evaluation_test still handle?

---

I think both evaluation test and rollout processors should update rollout status but the error handling should all be in evaluation test

---

what if you return a list of asyncio tasks for the rollout

---

i think that works for my case, but might mess with yinghan's case. lemme make the changes then ping him.